includes/LogisticsOrderFactory.php

```diff
@@ -9,21 +9,151 @@
 use Towa\GebruederWeissSDK\Model\LogisticsOrder;
+use Towa\GebruederWeissSDK\Model\LogisticsRequirements;
+use Towa\GebruederWeissSDK\Model\OrderLine;
 
 /**
  * Factory for creating Logistics Orders
  */
 class LogisticsOrderFactory
 {
+    /**
+     * Settings Repository
+     *
+     * @var SettingsRepository
+     */
+    private $settingsRepository;
+
+    /**
+     * Creates an instance of the logistics order factory
+     *
+     * @param SettingsRepository $settingsRepository An instance of the settings repository.
+     */
+    public function __construct(SettingsRepository $settingsRepository)
+    {
+        $this->settingsRepository = $settingsRepository;
+    }
+
     /**
      * Creates a logistics order from a WooCommerce order
      *
      * @param object $wooCommerceOrder The order to be converted into a logistics order.
      * @return LogisticsOrder
      */
-    public static function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    public function buildFromWooCommerceOrder(object $wooCommerceOrder): LogisticsOrder
+    {
+        $logisticsOrder = new LogisticsOrder();
+        $logisticsOrder->setUrl($this->settingsRepository->getSiteUrl() . "/wp-json/gebrueder-weiss-woocommerce/v1/update/" . $wooCommerceOrder->get_id());
+        $logisticsOrder->setCreationDateTime($wooCommerceOrder->get_date_created());
+
+        $logisticsOrder->setLogisticsAddresses([
+            $this->createConsigneeAddress($wooCommerceOrder),
+            $this->createOrderByAddress()
+        ]);
+
+        $logisticsOrder->setOrderLines(
+            $this->createOrderLines($wooCommerceOrder)
+        );
+
+        return $logisticsOrder;
+    }
+
+    /**
+     * Creates the consignee address from a WooCommerce Order.
+     *
+     * @param object $wooCommerceOrder The WooCommerce order.
+     * @return LogisticsAddress
+     */
+    private function createConsigneeAddress(object $wooCommerceOrder): LogisticsAddress
+    {
+        $logisticsAddress = new LogisticsAddress();
+        $logisticsAddress->setAddressType("consignee");
+
+        $address = new Address();
+        $address->setName1($wooCommerceOrder->get_shipping_first_name());
+        $address->setName2($wooCommerceOrder->get_shipping_last_name());
+        $address->setName3($wooCommerceOrder->get_shipping_company());
+        $address->setStreet1($wooCommerceOrder->get_shipping_address_1());
+        $address->setStreet2($wooCommerceOrder->get_shipping_address_2());
+        $address->setCity($wooCommerceOrder->get_shipping_city());
+        $address->setZipCode($wooCommerceOrder->get_shipping_postcode());
+        $address->setCountryCode($wooCommerceOrder->get_shipping_country());
+        $address->setState($wooCommerceOrder->get_shipping_state());
+        $logisticsAddress->setAddress($address);
+
+        $fullName = implode(" ", array_filter([$wooCommerceOrder->get_shipping_first_name(), $wooCommerceOrder->get_shipping_last_name()]));
```

Comment:
why d'you need array_filter here? implode concats array elements and the array is defined with the square brackets

Reply:
The array filter removes null values from the array. If a person with a single name orders something, this prevents from a space beeing added at the end or beginning.